test/e2e/auth/projected_clustertrustbundle.go

```diff
@@ -42,88 +48,494 @@ import (
 	"github.com/onsi/ginkgo/v2"
 )
 
+const (
+	testSignerOneName = "test.test/signer-one"
+	testSignerTwoName = "test.test/signer-two"
+	aliveSignersKey   = "signer.alive=true"
+	deadSignersKey    = "signer.alive=false"
+	noSignerKey       = "no-signer"
+)
+
 var _ = SIGDescribe(feature.ClusterTrustBundle, feature.ClusterTrustBundleProjection, func() {
 	f := framework.NewDefaultFramework("projected-clustertrustbundle")
 	f.NamespacePodSecurityLevel = admissionapi.LevelBaseline
 
-	goodCert1 := mustMakeCertificate(&x509.Certificate{
-		SerialNumber: big.NewInt(0),
-		Subject: pkix.Name{
-			CommonName: "root1",
-		},
-		IsCA:                  true,
-		BasicConstraintsValid: true,
-	})
+	initCTBs, pemMapping := initCTBData()
 
-	goodCert1Block := string(mustMakePEMBlock("CERTIFICATE", nil, goodCert1))
+	ginkgo.JustBeforeEach(func(ctx context.Context) {
+		cleanup := mustInitCTBs(ctx, f, initCTBs)
+		ginkgo.DeferCleanup(cleanup)
+	})
 
 	ginkgo.It("should be able to mount a single ClusterTrustBundle by name", func(ctx context.Context) {
 
-		ctb1 := &certificatesv1alpha1.ClusterTrustBundle{
-			ObjectMeta: metav1.ObjectMeta{
-				Name: "ctb1",
+		for _, tt := range []struct {
+			name           string
+			ctbName        string
+			optional       *bool
+			expectedOutput []string
+		}{
+			{
+				name:           "name of an existing CTB",
+				ctbName:        "test.test.signer-one.4",
+				expectedOutput: expectedRegexFromPEMs(initCTBs[4].Spec.TrustBundle),
 			},
-			Spec: certificatesv1alpha1.ClusterTrustBundleSpec{
-				TrustBundle: goodCert1Block,
+			{
+				name:           "name of a CTB that does not exist + optional=true",
+				ctbName:        "does-not-exist.at.all",
+				optional:       ptr.To(true),
+				expectedOutput: []string{"content of file \"/var/run/ctbtest/trust-anchors.pem\": \n$"},
 			},
+		} {
+			pod := podForCTBProjection(v1.VolumeProjection{
+				ClusterTrustBundle: &v1.ClusterTrustBundleProjection{
+					Name:     &tt.ctbName,
+					Path:     "trust-anchors.pem",
+					Optional: tt.optional,
+				},
+			})
+
+			fileModeRegexp := getFileModeRegex("/var/run/ctbtest/trust-anchors.pem", nil)
+			expectedOutput := append(tt.expectedOutput, fileModeRegexp)
+
+			e2epodoutput.TestContainerOutputRegexp(ctx, f, "project cluster trust bundle", pod, 0, expectedOutput)
```

Comment:
@stlaz go ahead and squash and we can merge this.

One question I had is "how do we make sure that there is no extra unexpected CA bundle data in the output" -> perhaps that is covered well enough in unit tests.

Reply:
> One question I had is "how do we make sure that there is no extra unexpected CA bundle data in the output" -> perhaps that is covered well enough in unit tests.

I originally thought that I would test the CAs in the bundle here, too, but the ordering is not very e2e friendly, so I checked unit tests and those seem to test the behavior quite thoroughly, so I decided to stick with unit tests only for that part.